STM32: H7RSx skip clock and xspi config when using XiP

TD-JBL opened the pull request:
I am using a customized OpenBLT bootloader which configures the clock system and the XSPI peripheral, then a Zephyr app is executed from external flash. I have not used MCUboot yet, but the problematic should be the same: 
Zephyr must not change the clock path used for clocking the XSPI peripheral and must not change XSPI configuration nor set a VOLTAGE_SCALE not allowed for the already high cpu clock.

I have `CONFIG_XIP=y` set in my prj.conf, `CONFIG_STM32_APP_IN_EXT_FLASH` is automatically set because of  its `default $(DT_FLASH_PARENT_IS_XSPI)` value.

With the changes in this PR, the combination of bootloader and zephyr runs fine on the nucleo_h7s3l8. But I'm not sure if this is the best way to handle things, so I open this up for discussion...

TD-JBL committed:
drivers: clk: stm32_ll_h7: Add stm32_clock_control_get_status function

Add stm32_clock_control_get_status() function needed by flash_stm32_xspi
driver for handling XiP.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

TD-JBL committed:
soc: st: stm32: h7rsx: Disable voltage scaling config during XiP

Disables setting the VOLTAGE_SCALE1 (VOS1) during soc_early_init_hook()
when CONFIG_STM32_APP_IN_EXT_FLASH is set.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

TD-JBL commented on drivers/clock_control/clock_stm32_ll_h7.c:
Is this a realistic usecase? Chainloading into an app in the small 64k internal flash next to the bootloader (not using XiP)? 
Guard with `#if !defined(CONFIG_STM32_APP_IN_EXT_FLASH)` or remove block?

etienne-lms commented on drivers/clock_control/clock_stm32_ll_h7.c:
```diff
@@ -680,10 +680,36 @@ static int stm32_clock_control_get_subsys_rate(const struct device *clock,
+
+	ARG_UNUSED(dev);
+
+	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX) == true) {
```
```suggestion
	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX)) {

etienne-lms commented on soc/st/stm32/stm32h7rsx/soc.c:
```diff
@@ -59,6 +61,8 @@ void soc_early_init_hook(void)
 	while (LL_PWR_IsActiveFlag_VOSRDY() == 0) {
 	}
 
+#endif /* defined(CONFIG_STM32_APP_IN_EXT_FLASH) */
```
```suggestion
#endif /* !CONFIG_STM32_APP_IN_EXT_FLASH */

TD-JBL committed:
boards: arm: st: nucleo_h7s3l8: Add *-block-size properties to xspi2

Add write-block-size and erase-block-size attributes to xspi nor flash
connected to xspi2. This is needed for mcuboot to handle flash access
of the ext_flash. The changes in the node structure is taken from the
stm32h7s78_dk board.

Update zephyr,flash-controller in samples/sysbuild/with_mcuboot/boards
/nucleo_h7s3l8.overlay to have CONFIG_STM32_APP_IN_EXT_FLASH set when
direct parent of flash is not XSPI but only the parent of the flash-
controller.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

TD-JBL committed:
drivers: flash: stm32_xspi: Remove hard-coded XSPI2 dependency

Check the used XSPIx peripheral before calling the corresponding function
instead of hard-coded calling the LL_PWR_EnableXSPIM2() and
LL_SBS_EnableXSPI2SpeedOptim() during flash_stm32_xspi_init(...).

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

TD-JBL committed:
dts: arm: st: h7rs: Enable HSI internal clock

Enable HSI clock node to fix get_hclk_frequency() returns 0 when called
from stm32_clock_control_init(). HSI on is the devices reset default
value anyway, but the STM32_HSI_FREQ define is set to 0 when not enabled.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

TD-JBL committed:
drivers: clock: stm32_ll_h7: Disable clock configuration for XiP

Extent set_up_plls() to check if we are running the app in XiP and only
reconfigure PLLs and PLL outputs when they are not used in the clock path
of Q/O/X-SPI or have other constraints like XSPI AXI clock >= XSPI kernel
clock.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>
